src/hooks/screen/useChatScreenHook.ts

```diff
@@ -378,8 +374,16 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
 
   React.useEffect(() => {
     if (localDb && selectedChannel) {
+      setChannelAsRead(selectedChannel, true);
       initChatData();
     }
+  }, [localDb, selectedChannel]);
+
+  React.useEffect(() => {
+    if (localDb && selectedChannel) {
+      setChannelAsRead(selectedChannel, true);
+      handleChatData();
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatData` function has been replaced with the newly introduced `handleChatData` function in the useEffect hook. This change should improve the efficiency of the code by reducing unnecessary data fetching operations.
```diff
-       initChatData();
+       handleChatData();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->